docs/dev-guide.md

```diff
@@ -255,3 +255,311 @@ which will properly configure and run all microservices.
+    3) "name"
+    4) "some"
+    5) "channels"
+    6) "ff13ca9c-7322-4c28-a25c-4fe5c7b753fc, c3642289-501d-4974-82f2-ecccc71b2d82, c3642289-501d-4974-82f2-ecccc71b2d83, cd4ce940-9173-43e3-86f7-f788e055eb14"
+    7) "externalID"
+    8) "9c:b6:d:eb:9f:fd"
+    9) "content"
+   10) "{}"
+   11) "timestamp"
+   12) "1555404899"
+   13) "operation"
+   14) "config.create"
+   15) "id"
+   16) "63a110d4-2b77-48d2-aa46-2582681eeb82"
+```
+
+#### Configuration update event
+Whenever configuration is updated, `bootstrap` service will generate and publish
+new create event. This event will have the following format:
+```
+1) "1555405104368-0"
+2)  1) "content"
+    2) "NOV_MGT_HOST: http://127.0.0.1:7000\nDOCKER_MGT_HOST: http://127.0.0.1:2375\nAGENT_MGT_HOST: https://127.0.0.1:7003\nMF_MQTT_HOST: tcp://104.248.142.133:8443"              
+    3) "timestamp"
+    4) "1555405104"
+    5) "operation"
+    6) "config.update"
+    7) "id"
+    8) "63a110d4-2b77-48d2-aa46-2582681eeb82"
+    9) "name"
+   10) "weio"
+```
+
+#### Configuration remove event
+Whenever configuration is removed, `bootstrap` service will generate and publish
+new remove event. This event will have the following format:
+```
+1) "1555405464328-0"
+2) 1) "id"
+   2) "63a110d4-2b77-48d2-aa46-2582681eeb82"
+   3) "timestamp"
+   4) "1555405464"
+   5) "operation"
+   6) "config.remove"
+```
+
+#### Thing bootstrap event
+Whenever thing is bootstrapped, `bootstrap` service will generate and publish
+new bootstrap event. This event will have the following format:
+```
+1) "1555405173785-0"
+2) 1) "externalID"
+   2) "9c:b6:d:eb:9f:fd"
+   3) "successful"
+   4) "1"
+   5) "timestamp"
+   6) "1555405173"
+   7) "operation"
+   8) "thing.bootstrap"
+```
+
+#### Thing change state event
+Whenever thing's state changes, `bootstrap` service will generate and publish
+new change state event. This event will have the following format:
+```
+1) "1555405294806-0"
+2) 1) "id"
+   2) "63a110d4-2b77-48d2-aa46-2582681eeb82"
+   3) "state"
+   4) "0"
+   5) "timestamp"
+   6) "1555405294"
+   7) "operation"
+   8) "thing.state_change"
+```
+
+#### Thing update connections event
+Whenever thing's list of connections is updated, `bootstrap` service will generate
+and publish new update connections event. This event will have the following format:
+```
+1) "1555405373360-0"
+2) 1) "operation"
+   2) "thing.update_connections"
+   3) "id"
+   4) "63a110d4-2b77-48d2-aa46-2582681eeb82"
+   5) "channels"
+   6) "ff13ca9c-7322-4c28-a25c-4fe5c7b753fc, 925461e6-edfb-4755-9242-8a57199b90a5, c3642289-501d-4974-82f2-ecccc71b2d82"
+   7) "timestamp"
+   8) "1555405373"
+```
+
+### MQTT Adapter
+Instead of using heartbeat to know when client is connected through MQTT adapter one
+can fetch events from Redis Streams that MQTT adapter publishes. MQTT adapter
+publishes events every time client connects and disconnects to stream named `mainflux.mqtt`.
+
+Events that are coming from MQTT adapter have following field:
```

Comment:
comming

Reply:
fields

Reply:
What `comming`? You're reviewing only first commit. Please don't review commits but last PR state.

Reply:
my bad